sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -512,49 +512,27 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * For Hive metastore table, the metadata is refreshed. For data source tables, the schema will
    * not be inferred and refreshed.
    *
-   * If this table is cached as an InMemoryRelation, drop the original cached version and make the
-   * new version cached lazily.
+   * If this table is cached as an InMemoryRelation, re-cache the table and its dependents lazily.
    *
-   * In addition, refreshing a table also invalidate all caches that have reference to the table
+   * In addition, refreshing a table also clear all caches that have reference to the table
```

Comment:
This seems duplicated with the above statement `re-cache the table and its dependents lazily.`. Maybe we should change the above to `re-cache the table lazily.`

Reply:
Don't think it is a duplicate. It describes the case when a table is not cached. See the PR https://github.com/apache/spark/pull/30187 in which the statement was added by @sunchao and committed by @dongjoon-hyun .